src/connector/src/sink/writer.rs

```diff
@@ -113,13 +113,16 @@ pub trait FormattedSink {
 pub struct LogSinkerOf<W> {
     writer: W,
     sink_metrics: SinkMetrics,
+
+    commit_checkpoint_interval: u64,
```

Comment:
We'd better implement this special logic in the iceberg sink writer and leave this shared default implementation unchanged. We can implement a new `IcebergLogSinker` to consume the log, and then for every `commit_checkpoint_interval` number of checkpoint barrier we then commit to iceberg and then truncate the log reader.